packages/vue-client/src/views/GameView.vue

```diff
@@ -146,40 +146,39 @@ function makeMove(move_str: string) {
   }
 
   move[playing_as.value] = move_str;
-  requests
-    .post(`/games/${gameResponse.id}/move`, move)
-    .then((res: GameResponse) => {
-      Object.assign(gameResponse, res);
-    })
-    .catch(alert);
+  requests.post(`/games/${props.gameId}/move`, move).catch(alert);
 }
 
 watchEffect((onCleanup) => {
   if (!props.gameId) {
     return;
   }
-  const message = `game/${props.gameId}`;
-  socket.on(message, (data) => {
-    Object.assign(gameResponse, data);
-  });
+  const topic = `game/${props.gameId}`;
+  socket.emit("subscribe", [topic]);
+  socket.on("move", (state: GameStateResponse) =>
+    setViewState(state.round, state.seat),
```

Comment:
Shouldn't we call `setNewState` directly?  `setViewState` will trigger another API call, but that's not necessary since we already have a `GameStateResponse`.

Reply:
You are right, thank you for catching this!